wrappers/desktop.go

```diff
@@ -247,7 +247,45 @@ func findDesktopFiles(rootDir string) ([]string, error) {
 	return desktopFiles, nil
 }
 
-func deriveDesktopFilesContent(s *snap.Info) (map[string]osutil.FileState, error) {
+func findSnapDesktopFileIDs(s *snap.Info) (map[string]bool, error) {
+	var desktopPlug *snap.PlugInfo
+	for _, plug := range s.Plugs {
+		if plug.Interface == "desktop" {
+			desktopPlug = plug
+			break
+		}
+	}
+	if desktopPlug == nil {
+		return nil, nil
+	}
+
+	attrVal, exists := desktopPlug.Lookup("desktop-file-ids")
+	if !exists {
+		// desktop-file-ids attribute is optional
+		return nil, nil
```

Comment:
This check and the type checks below aren't tested. May be worth a short table test

Reply:
Nice catch, added tests